openeo/extra/job_management/__init__.py

```diff
@@ -535,13 +560,18 @@ def _job_update_loop(
                     to_add = self.backends[backend_name].parallel_jobs - backend_load
                     for i in not_started.index[total_added : total_added + to_add]:
                         self._launch_job(start_job, df=not_started, i=i, backend_name=backend_name, stats=stats)
-                        stats["job launch"] += 1
-
                         job_db.persist(not_started.loc[i : i + 1])
+
+                        stats["job launch"] += 1
                         stats["job_db persist"] += 1
                         total_added += 1
 
         # Act on jobs
+        # TODO: move this back closer to the `_track_statuses` call above, once job done/error handling is also handled in threads?
```

Comment:
This todo is about `for job, row in jobs_done ...`  not about `_process_result_queue`